### PR TITLE
chore(performance): move more sources to send_batch

### PR DESCRIPTION
**Pull request by lukesteensen**
Signed-off-by: Luke Steensen <luke.steensen@gmail.com>

### CODE REVIEW

**Review by tobz — APPROVED** [review]

**Review by 001wwang — APPROVED** [review]
Neat!

File: src/source_sender/mod.rs [hunk 1]
```diff
@@ -174,6 +174,18 @@ impl SourceSender {
             .await
     }
 
+    pub async fn send_batch_named(
+        &mut self,
+        name: &str,
+        events: Vec<Event>,
+    ) -> Result<(), ClosedError> {
+        self.named_inner
```
[the remark this answers is not in the document]
Comment (lukesteensen) [reply src/source_sender/mod.rs]: This is what I get for assuming my tiny patch will obviously compile 😬

**Review by bruceg — APPROVED** [review]
LGTM and a big :+1: on removing the `stream::iter` just to submit a batch.

**Comment by bruceg** [comment]
Would there be any opportunity to move `send_batch` to an interface that took, for example, `impl Iterator<Item = impl Into<Event>>` and do the `collect` inside that interface? That would play very well into the array work instead of making it reshape the current `Vec<Event>` into one or more arrays. That probably would best be done as a follow-on issue.

**Pull request merged**